ksp_plugin_adapter/ksp_plugin_adapter.cs

```diff
@@ -264,10 +269,11 @@ private enum UnityLayers {
                 "; this build targets " + expected_version + ".");
     }
     map_node_pool_ = new MapNodePool();
-  }
-
-  ~PrincipiaPluginAdapter() {
-    Cleanup();
```

Comment:
The `PrincipiaPluginAdapter` does have ownership of an unmanaged resource, namely the plugin. It would make sense to destroy it here if it is not IntPtr.Zero (perhaps there are cases where the normal lifecycle of the game is interrupted, and the object finalized without ever going through OnDisable).